### ISSUE TITLE
Hacking CloudKit - How I accidentally deleted your Apple Shortcuts - Detectify Labs

### ISSUE THREAD

**Issue by dropob** [dropod/issues#13377]
TL;DR. CloudKit, the data storage framework by Apple, has various access controls. These access controls could be misconfigured, even by Apple themselves, which affected Apple’s own apps using CloudKit. This blog post explains in detail three bugs found in iCrowd+, Apple News and Apple Shortcuts with different criticality uncovered by Frans Rosen while hacking Cloudkit. All bugs were reported to and fixed by the Apple Security Bounty program.
Intro
Ever since the fantastic “We Hacked Apple for 3 ...

**[archive.today](https://archive.today/)**:
> source: [https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/](https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/)
> archived: [http://archive.today?url=https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/](http://archive.today?url=https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/)

**[IPFS](https://ipfs.github.io/public-gateway-checker/)**:
> source: [https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/](https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/)
> archived: [https://ipfs.io/ipfs/bafybeih4cldrkjk5koimpcpekdz6obaduweerrr47ksqjrva3j3sbqtmgy](https://ipfs.io/ipfs/bafybeih4cldrkjk5koimpcpekdz6obaduweerrr47ksqjrva3j3sbqtmgy)

**[Telegraph](https://telegra.ph/)**:
> source: [https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/](https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/)
> archived: create page failed: CONTENT_TOO_BIG

**[Internet Archive](https://web.archive.org/)**:
> source: [https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/](https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/)
> archived: [https://web.archive.org/web/20220201034121/https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/](https://web.archive.org/web/20220201034121/https://labs.detectify.com/2021/09/13/hacking-cloudkit-how-i-accidentally-deleted-your-apple-shortcuts/)

**[AnonFiles](https://anonfiles.com/)** - [ [IMG](https://anonfiles.com/33U3Z6Edxe) ¦ [PDF](https://anonfiles.com/54UfZ7E2x9) ¦ [RAW](https://anonfiles.com/z0U3Z9Efx5) ¦ [TXT](https://anonfiles.com/x8UfZ9Efxe) ¦ [HAR](https://anonfiles.com/14UaZ8E7x4) ¦ [WARC](https://anonfiles.com/96U9ZbEbx7) ¦ [MEDIA](https://anonfiles.com/73UaZ3E9xe) ]
**[Catbox](https://catbox.moe/)** - [ [IMG](https://files.catbox.moe/gnuro8.png) ¦ [PDF](https://files.catbox.moe/z7jyd7.pdf) ¦ [RAW](https://files.catbox.moe/eu8lgu.htm) ¦ [TXT](https://files.catbox.moe/sg0jm3.txt) ¦ [HAR](https://files.catbox.moe/hsxnoe.har) ¦ [WARC](https://files.catbox.moe/0ypqrg.warc) ¦ [MEDIA](https://files.catbox.moe/0vvdw0.mp4) ]
